tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/WhatisResponse.java

```diff
@@ -0,0 +1,65 @@
+package org.unicode.cldr.web.api;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.unicode.cldr.util.CLDRLocale;
+
+public final class WhatisResponse {
+    @Schema(description = "query string")
+    public String q;
+    public String q2 = null;
+    public String err = null;
+    public List<XpathResult> xpath = null;
+    public List<StandardCodeResult> exact = null;
+    public List<StandardCodeResult> full = null;
```

Comment:
I'm getting this when exact is non-null and non-empty:
```
[ERROR   ] Unable to serialize property 'exact' from org.unicode.cldr.web.api.WhatisResponse
[ERROR   ] Generating incomplete JSON
[ERROR   ] Problem with writing the data, class org.unicode.cldr.web.api.WhatisResponse, ContentType: application/json
[err] java.lang.Throwable: ANDY
```

Reply:
yes ANDY commented here https://github.com/OpenLiberty/open-liberty/issues/15845#issuecomment-791722057

What you need to do is to add a `public StandardCodeResult[] getExact() {}` function,  using `toArray` - see for example: 

https://github.com/unicode-org/cldr/pull/1090/files#diff-1d25604a4ccc92b62c7dfc8f79474238ce51a65ac3812e38ce663bacfb4a5b4aR854-R858

a public `getExact()` function causes `exact` to show up as a property.

Reply:
I don't actually see what you're referring to in that PR; anyway I'll try adding `getExact()`, thanks!

It's mysterious why or when a getter is needed -- converting a Java List to a json array would seem like a natural default. If it takes so much code to convert an object to json, then the advantages over using JSONObject aren't so clear... Do you happen to know a good guide to using javax.ws.rs.core.Response? Other than https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/Response.html ?

Reply:
I don't know off hand no

Reply:
```java
       List<ReviewEntry> entries;

        // for serialization
         public ReviewEntry[] getEntries() {
             return entries.toArray(new ReviewEntry[entries.size()]);
         }
```

Reply:
I think the situation is that `List` isn't a  primitive JSON type ,and won't round-trip from an array.

Reply:
Here's my latest WhatisResponse.java:
```
package org.unicode.cldr.web.api;

import java.util.ArrayList;
import java.util.List;

import org.eclipse.microprofile.openapi.annotations.media.Schema;
import org.unicode.cldr.util.CLDRLocale;

public final class WhatisResponse {
    @Schema(description = "query string")
    public String q;
    public String q2 = null;
    public String err = null;
    public List<XpathResult> xpath = null;
    public List<StandardCodeResult> exact = null;
    public List<StandardCodeResult> full = null;
    public int pathsSearched = 0;
    public int localesSearched = 0;

    public WhatisResponse(String queryString) {
        q = queryString;
    }

    public XpathResult[] getXpath() {
        System.out.println("Hello my name is getXpath");
        return xpath.toArray(new XpathResult[xpath.size()]);
    }

    public StandardCodeResult[] getExact() {
        System.out.println("Hello my name is getExact");
        return exact.toArray(new StandardCodeResult[exact.size()]);
    }

    public StandardCodeResult[] getFull() {
        System.out.println("Hello my name is getFull");
        return full.toArray(new StandardCodeResult[full.size()]);
    }

    class XpathResult {
        public String loc;
        public String name;
        public List<PathAndValue> matches = new ArrayList<>();

        public XpathResult(CLDRLocale locale) {
            loc = locale.toString();
            name = locale.getDisplayName();
        }

        public PathAndValue[] getMatches() {
            System.out.println("Hello my name is getMatches");
            return matches.toArray(new PathAndValue[matches.size()]);
        }
    }

    class PathAndValue {
        public String path;
        public String value;

        public PathAndValue(String p, String v) {
            path = p;
            value = v;
        }
    }

    class StandardCodeResult {
        public String type;
        public String code;
        public List<StandardCodeResultDataItem> data = new ArrayList<>();

        public StandardCodeResult(String t, String c) {
            type = t;
            code = c;
        }

        public StandardCodeResultDataItem[] getData() {
            System.out.println("Hello my name is getData");
            return data.toArray(new StandardCodeResultDataItem[data.size()]);
        }
    }

    class StandardCodeResultDataItem {
        public String s;
        public boolean isWinner;

        public StandardCodeResultDataItem(String str, boolean win) {
            s = str;
            isWinner = win;
        }
    }
}
```

And here's the console:

```
Hello my name is getExact
[ERROR   ] Unable to serialize property 'exact' from org.unicode.cldr.web.api.WhatisResponse
[ERROR   ] Generating incomplete JSON
[ERROR   ] Problem with writing the data, class org.unicode.cldr.web.api.WhatisResponse, ContentType: application/json
[err] java.lang.Throwable: ANDY
[err] 	at org.apache.cxf.jaxrs.utils.JAXRSUtils.logMessageHandlerProblem(JAXRSUtils.java:2055)
[err] 	at [internal classes]
[err] 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:686)
[err] 	at com.ibm.websphere.jaxrs.server.IBMRestServlet.service(IBMRestServlet.java:96)
[err] 	at [internal classes]
[err] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
[err] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
[err] 	at java.base/java.lang.Thread.run(Thread.java:853)
[WARNING ] Interceptor for {http://api.web.cldr.unicode.org/}About has thrown exception, unwinding now
Unable to serialize property 'exact' from org.unicode.cldr.web.api.WhatisResponse
[WARNING ] Could not create a secure Stax XMLInputFactory.  Found class com.sun.xml.internal.stream.XMLInputFactoryImpl.  Suggest Woodstox 4.2.0 or newer.
```

I have no idea whether `Woodstox` is related...

Reply:
don't make these public. 
```java
    public List<StandardCodeResult> exact = null;
    public List<StandardCodeResult> full = null;
```

Reply:
> don't make these public.

I changed them to private. The `getExact` function does get called, and then the same error still occurs: `Unable to serialize property 'exact'.`